fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/Button.kt

```diff
@@ -92,80 +97,82 @@ fun Button(
         borderModifier = borderModifier.border(borderWidth, border.brush, shape)
     }
 
-    Box(
-        modifier
-            .heightIn(min = token.fixedHeight(buttonInfo))
-            .background(
-                brush = backgroundColor,
-                shape = shape
-            )
-            .clip(shape)
-            .semantics(true) {
-                editableText = AnnotatedString(text ?: "")
-                this.contentDescription = contentDescription ?: ""
-            }
-            .then(clickAndSemanticsModifier)
-            .then(borderModifier),
-        propagateMinConstraints = true
-    ) {
-        Row(
-            Modifier.padding(contentPadding),
-            horizontalArrangement = Arrangement.spacedBy(
-                iconSpacing,
-                Alignment.CenterHorizontally
-            ),
-            verticalAlignment = Alignment.CenterVertically
-        ) {
-
-            if (icon != null)
-                Icon(
-                    imageVector = icon,
-                    contentDescription = null,
-                    modifier = Modifier
-                        .size(
-                            token.iconSize(buttonInfo = buttonInfo)
-                        ),
-                    tint = token.iconColor(buttonInfo = buttonInfo)
-                        .getColorByState(
-                            enabled = enabled,
-                            selected = false,
-                            interactionSource = interactionSource
-                        )
+    Surface(elevation = elevation, shape = shape, modifier = modifier) {
```

Comment:
instead of using surface, can we try shadow modifier?

Reply:
the order in which we apply the shadow modifier changes the effect 
Refer this link
https://stackoverflow.com/questions/75950471/how-to-make-rounded-corner-with-shadow-in-jetpack-compose

Reply:
Surface makes it easier overall and ensures that adding any new child will never break the elevation and shadows. Plus allows for cleaner code as well.